lib/batch/institutional_nucats_template.rb

```diff
@@ -0,0 +1,40 @@
+# https://github.com/galterlibrary/digital-repository/issues/691
+# Create 10 'template' institutional collections, adding another 3 template
+# collections to each. For more immediate use when needed.
+
+# NUCATS Grants Repository
+parent_nucats_grants = Collection.find("f2bf6e1d-0e32-4ce2-a52e-bb0522d5708d")
+
+10.times do |i|
+  template = Collection.new(
+    title: "template-#{i}", 
+    tag: ["grant proposals", "research support", "peer-reviewed grants"]
+  )
+  template.apply_depositor_metadata(parent_nucats_grants.depositor)
+  template.save!
+
+  3.times do |child_i|
+    child = Collection.new(
+      title: "child-#{child_i}-of-template-#{i}",
+      tag: ["grant proposals", "research support", "peer-reviewed grants"]
+    )
+    child.apply_depositor_metadata(parent_nucats_grants.depositor)
+    child.save!
+
+    template.members << child
+    template.save!
+
+    # because we have a before create action that sets visibility to "open"
+    child.update_attributes(visibility: "restricted")
```

Comment:
Do you have to `save!` after this? or does this call `save!` behind the scene in which case you don't need to `save!` above?

Reply:
If I remove the `save!` above, the visibility will be set to "open" here because the collection is still not yet created. Once the collection is created, the `update_attributes` will save any changes. I verified this by testing locally.